src/java/org/apache/cassandra/service/ReplicaFilteringProtection.java

```diff
@@ -239,7 +253,21 @@ public Row onMergedRows(Row merged, Row[] versions)
                 {
                     // cache the row versions to be able to regenerate the original row iterator
                     for (int i = 0; i < versions.length; i++)
+                    {
                         builders[i].addRow(versions[i]);
+                        rowsCached++;
+
+                        if (rowsCached == cachedRowsFailThreshold)
```

Comment:
I wonder if it would be better to use `>` instead of `==`. If we used `>` and we were counting merged materialized/cached rows instead of per replica, we could guarantee that, without conflicts, a threshold of N rows allows a max limit of N rows. Also, I think that `tombstone_failure_threshold` uses `>`, so it will be more consistent with that meaning of threshold value. WDYT?

Reply:
The main reason why I used `==` here was to make sure we only logged the warning once. I could just as easily make this `rowsCached == cachedRowsFailThreshold + 1`. You're right that "threshold" usually means the level that must be _exceeded_ for some action to take place :)